server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java

```diff
@@ -476,7 +476,7 @@ private static Set<String> getPinnedTimestampLockedFiles(
         Set<Long> newPinnedTimestamps = new TreeSet<>(Collections.reverseOrder());
         for (Long pinnedTimestamp : pinnedTimestampSet) {
             String cachedFile = metadataFilePinnedTimestampMap.get(pinnedTimestamp);
-            if (cachedFile != null) {
+            if (cachedFile != null && metadataFiles.contains(cachedFile)) {
```

Comment:
Extra check to make sure that returned list of locked metadata files will always be a sub-set of provided list of metadata files.

Reply:
Makes sense. Let me add assert as well.

Reply:
@gbbafna Do you mean removing the check and adding it as an assert?

Reply:
Made the change